File: public/lib/fidl/go/src/fidl/compiler/generators/go/templates/enums.go
// Copyright 2016 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package templates

import (
	"text/template"
)

const enumDeclTmplText = `
{{- define "EnumDecl" -}}
{{$enum := . -}}
type {{$enum.Name}} int32

{{ template "RuntimeTypeAccessors" $enum }}

const (
{{- range $value := $enum.Values -}}
	{{$value.Name}} {{$enum.Name}} = {{$value.Value}}
{{end}}
)

{{- end -}}
`

func initEnumTemplates() {
	template.Must(goFileTmpl.Parse(enumDeclTmplText))
}

File: tools/fidl/difl/enums.py
# Copyright 2019 The Fuchsia Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.

from typing import List, Optional, Sequence, Tuple, FrozenSet, Set, Dict, NamedTuple

from difl.ir import Library, Struct, StructMember, Type, Table, Union, Protocol, Enum
from difl.changes import *
from difl.comparator import Comparator


def enum_changes(before: Enum, after: Enum,
                 comparator: Comparator) -> List[Change]:
    changes: List[Change] = []

    # TODO: actually compare enums

    return changes
